test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,461 @@
+
+@dataclasses.dataclass
+class XfailOpset:
+    """Expects a OpInfo test to fail on specific ONNX opsets."""
+
+    op_name: str
+    opsets: Collection[Union[int, Callable[[int], bool]]]
+    dtypes: Optional[Collection[torch.dtype]] = None
+    exception: Optional[Exception] = None
+    reason: str = "unspecified"
+
+    def _contains_opset(self, opset: int) -> bool:
+        return any(
+            opset == opset_spec if isinstance(opset_spec, int) else opset_spec(opset)
+            for opset_spec in self.opsets
+        )
+
+    def _contains_dtype(self, dtype: torch.dtype) -> bool:
+        return self.dtypes is None or dtype in self.dtypes
+
+    def should_fail(self, opset: int, dtype: torch.dtype) -> bool:
+        """Returns whether the test should fail for the given opset and dtype."""
+        return self._contains_opset(opset) and self._contains_dtype(dtype)
+
+
+def skip_ops(
+    all_opinfos: Sequence[opinfo_core.OpInfo],
+    test_case_name: str,
+    base_test_name: str,
+    skip_or_xfails: Iterable[DecorateMeta],
+):
+    """Decorates OpInfo tests with decorators based on the skip_or_xfails list."""
+    ops_mapping = {(info.name, info.variant_test_name): info for info in all_opinfos}
+    for decorate_meta in skip_or_xfails:
+        opinfo = ops_mapping.get((decorate_meta.op_name, decorate_meta.variant_name))
+        assert (
+            opinfo is not None
+        ), f"Couldn't find OpInfo for {decorate_meta}. Did you need to specify variant_name?"
+        decorators = list(opinfo.decorators)
+        new_decorator = opinfo_core.DecorateInfo(
+            decorate_meta.decorator,
+            test_case_name,
+            base_test_name,
+            device_type=decorate_meta.device_type,
+            dtypes=decorate_meta.dtypes,
+        )
+        decorators.append(new_decorator)
+        opinfo.decorators = tuple(decorators)
+
+    # This decorator doesn't modify fn in any way
+    def wrapped(fn):
+        return fn
+
+    return wrapped
+
+
+def get_torch_op_name(op: Union[str, Callable]) -> str:
+    """Returns the name of the torch function corresponding to the given op."""
+    if callable(op):
+        module_name = op.__module__.split("torch.", 1)
+        op_name = op.__name__
+        if len(module_name) == 2:
+            # Remove the torch. prefix
+            op_name = f"{module_name[1]}.{op_name}"
+        return op_name
+    # Already a string
+    return op
+
+
+def opsets_before(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is before the specified."""
+
+    def compare(other_opset: int):
+        return other_opset < opset
+
+    return compare
+
+
+def opsets_after(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is after the specified."""
+
+    def compare(other_opset: int):
+        return other_opset > opset
+
+    return compare
+
+
+def reason_onnx_runtime_does_not_support(
```

Comment:
Discussed with @justinchuby. There are common cases that valid export fails this test because that particular dtype was not implemented in ort. @AllenTiTaiWang let's see if we can utilize onnx-script and its reference implementation.